### DISCUSSION TITLE
EJBCA CE 7.10.0.2 and utf-8 error while downloading PEM file from RA

### DISCUSSION THREAD

**Discussion by vladex42** [Keyfactor/ejbca-ce#174] [General]
Hi all :)

I'm testing EJBCA CE latest version and I have encountered an interesting issue while trying to download PEM file from RA interface.
If I put any of šđčćž chars in CN of user, I get an error like below in Apache logs.

```
[Thu Nov 24 12:01:11.348883 2022] [http:error] [pid 59091:tid 139925672392448] [client 127.0.0.1:41690] AH02430: Response header 'Content-Disposition' value of 'attachment; filename="Stevo `tefanovrix07.pem"' contains invalid characters, aborting request, referer: https://localhost/ejbca/ra/enrollwithusername.xhtml
[Thu Nov 24 12:01:11.349075 2022] [proxy_ajp:error] [pid 59091:tid 139925672392448] [client 127.0.0.1:41690] AH02822: dialog with client 127.0.0.1:41690 failed, referer: https://localhost/ejbca/ra/enrollwithusername.xhtml
```

Anybody encountered similar issue and found a fix for this?
Ofc, thing works if I remove šđčćž from CN and replace them with sdcz.
I have tried to fix Apache, AJP module, Wildfly, MariaDB with all UTF-8 options to no success.

Thanx!

**Comment by mike-agrenius-kushner** [comment]
Hi, 

Yeah, we're super restrictive about what characters we allow to use, and it's the ` that's giving you problems. We've planned making it a bit more user friendly, but you should be able to get by escaping it with a '\'

Cheers,
Mike
